src/DGtal/geometry/surfaces/estimation/PlaneProbingR1Neighborhood.h

```diff
@@ -0,0 +1,227 @@
+// Inclusions
+#include <iostream>
+#include "DGtal/base/Common.h"
+#include "DGtal/geometry/helpers/PlaneProbingEstimatorHelper.h"
+#include "DGtal/geometry/surfaces/estimation/PlaneProbingRNeighborhood.h"
+#include "DGtal/kernel/CPointPredicate.h"
+//////////////////////////////////////////////////////////////////////////////
+
+namespace DGtal
+{
+
+  /////////////////////////////////////////////////////////////////////////////
+  // template class PlaneProbingR1Neighborhood
+  /**
+   * Description of template class 'PlaneProbingR1Neighborhood' <p>
+   * \brief Aim: Represent a way to probe the R-neighborhood, with the R1 optimization,
+   * see \cite RLDGCI2019 for details.
+   *
+   * \tparam TPredicate the probing predicate, a model of concepts::CPointPredicate.
+   */
+  template <typename TPredicate>
+  class PlaneProbingR1Neighborhood : public PlaneProbingRNeighborhood<TPredicate>
+  {
+    BOOST_CONCEPT_ASSERT((concepts::CPointPredicate<TPredicate>));
+
+    // ----------------------- Public types ------------------------------
+  public:
+      using Predicate    = TPredicate;
+      using Point        = typename TPredicate::Point;
+      using Triangle     = typename PlaneProbingRNeighborhood<TPredicate>::Triangle;
+      using ProbingRay   = typename PlaneProbingRNeighborhood<TPredicate>::ProbingRay;
+      using Integer      = typename Point::Coordinate;
+      using HexagonState = typename PlaneProbingRNeighborhood<TPredicate>::HexagonState;
+
+    // ----------------------- Standard services ------------------------------
+  public:
+    /**
+     * Default constructor.
+     */
+    PlaneProbingR1Neighborhood() = delete;
+
+    /**
+     * Constructor.
+     *
+     * @param aPredicate a probing predicate.
+     * @param aQ the fixed point 'q'.
+     * @param aM a frame composed of the three vectors.
+     */
+    PlaneProbingR1Neighborhood(Predicate const& aPredicate, Point const& aQ, Triangle const& aM);
+
+    /**
+     * Destructor.
+     */
+    ~PlaneProbingR1Neighborhood();
+
+    /**
+     * Copy constructor.
+     * @param other the object to clone.
+     */
+    PlaneProbingR1Neighborhood ( const PlaneProbingR1Neighborhood & other ) = delete;
+
+    /**
+     * Move constructor.
+     * @param other the object to move.
+     */
+    PlaneProbingR1Neighborhood ( PlaneProbingR1Neighborhood && other ) = delete;
+
+    /**
+     * Copy assignment operator.
+     * @param other the object to copy.
+     * @return a reference on 'this'.
+     */
+    PlaneProbingR1Neighborhood & operator= ( const PlaneProbingR1Neighborhood & other ) = delete;
+
+    /**
+     * Move assignment operator.
+     * @param other the object to move.
+     * @return a reference on 'this'.
+     */
+    PlaneProbingR1Neighborhood & operator= ( PlaneProbingR1Neighborhood && other ) = delete;
+
+    // ----------------------- Plane Probing services ------------------------------
+  public:
+    HexagonState hexagonState ();
```

Comment:
Same as above, add "virtual" and "override" where it applies.

Reply:
Fixed.